qa/tasks/cephfs/test_misc.py

```diff
@@ -142,6 +142,39 @@ def test_evict_client(self):
         ls_data = self.fs.mds_asok(['session', 'ls'])
         self.assert_session_count(1, ls_data)
 
+    def test_dont_mark_hanged_client_stale(self):
+        """
+        Test that a client holding no caps in a hanged state is not marked
+        stale after session_timeout and is evicted directly at
+        session_autoclose.
+
+        To conduct this test we need more that two clients since a single
+        client is never evcited by MDS.
+        """
```

Comment:
I'd prefer to see this test do:

(a) mkdir something on mount_a
(b) sigstop mount_a
(c) wait session timeout
(d) check it's not stale
(e) stat dir with mount_b
(f) confirm mount_a is marked stale
(g) wait (session_autoclose-session_timeout+2)
(h) mount_a is evicted and mount_b stat completes

Reply:
I went through the comments on this PR to make sure if I am not confused about this (since I do recall that at the very beginning I did write the code to mark sessions at stale). The decision to not to mark stale was made [here](https://github.com/ceph/ceph/pull/22645#discussion_r197319784).